### PR TITLE
LP Token Patch

### PR DESCRIPTION
**Pull request by maseca**
Addresses Cumulus bug that breaks Basic Auth for s3 credentials retrieval

### CODE REVIEW

**Comment by riverma** [comment]
@maseca - great work and timing here! Any changes specified to the `.netrc` file to leverage the token? Can you give a template example of how to specific the token (without placing any token here) in the `.netrc`?

**Comment by maseca** [comment]
So the token handling is done via the URS token API. The `.netrc` file is only used to provide the basic auth required to interact with the token API. Once the token is retrieved by the script via API it is stored in working memory until the script terminates. If no existing valid token is present based on the results of the initial API call, a second call will be performed in order to generate a valid token.

File: data_subscriber/daac_data_subscriber.py [hunk 1]
```diff
@@ -1252,10 +1252,11 @@ def _to_s3_url(dl_dict: dict[str, Any]) -> str:
 
 
 @ttl_cache(ttl=3300)  # 3300s == 55m. Refresh credentials before expiry. Note: validity period is 60 minutes
-def _get_aws_creds(session):
+def _get_aws_creds(token):
     logging.info("entry")
 
-    with session.get("https://data.lpdaac.earthdatacloud.nasa.gov/s3credentials") as r:
+    with requests.get("https://data.lpdaac.earthdatacloud.nasa.gov/s3credentials",
+                      headers={'Authorization': f'Bearer {token}'}) as r:
```
Comment (riverma) [inline data_subscriber/daac_data_subscriber.py:1259]: Nicely done!

**Review by riverma — COMMENTED** [review]
Great work @maseca - appreciate your rapid patch here. 

Questions:
- Will an operator need to specify the token explicitly somewhere? 
- How long would a token last, and is automatic renewal taken care of?

File: data_subscriber/daac_data_subscriber.py (hunk 1 above)
Comment (LucaCinquini) [reply data_subscriber/daac_data_subscriber.py:1259]: I agree - great surgical changes :)

**Comment by maseca** [comment]
Operators will still have access to the URS GUI for anything token-related, but they shouldn't have to manually do anything. The PCM will generate new tokens automatically via the API when they are about to expire.

**Review by hhlee445 — APPROVED** [review]

**Pull request merged**